File: Chapter4/Python/Example1.py
# Chapter 4 - Poisson Distribution
# Example 1: Basic Probability Calculations  Poisson(5)

from scipy.stats import poisson

# Parameter
lam = 5

# P(X = 4)
prob_exact = poisson.pmf(4, lam)
print(f"P(X = 4) = {prob_exact:.4f}")

# P(X <= 6)
prob_cumulative = poisson.cdf(6, lam)
print(f"P(X <= 6) = {prob_cumulative:.4f}")

# P(X >= 3) = 1 - P(X <= 2)
prob_greater = 1 - poisson.cdf(2, lam)
print(f"P(X >= 3) = {prob_greater:.4f}")

File: Chapter4/R/Example1.R
# Chapter 4 - Poisson Distribution
# Example 1: Basic Probability Calculations  Poisson(5)

# Parameter
lambda <- 5

# P(X = 4)
prob_exact <- dpois(4, lambda)
cat(sprintf("P(X = 4) = %.4f\n", prob_exact))

# P(X <= 6)
prob_cumulative <- ppois(6, lambda)
cat(sprintf("P(X <= 6) = %.4f\n", prob_cumulative))

# P(X >= 3) = 1 - P(X <= 2)
prob_greater <- 1 - ppois(2, lambda)
cat(sprintf("P(X >= 3) = %.4f\n", prob_greater))
